src/librustc/dep_graph/visit.rs

```diff
@@ -78,9 +79,30 @@ pub fn visit_all_item_likes_in_krate<'a, 'tcx, V, F>(tcx: TyCtxt<'a, 'tcx, 'tcx>
 pub fn visit_all_bodies_in_krate<'a, 'tcx, C>(tcx: TyCtxt<'a, 'tcx, 'tcx>, callback: C)
     where C: Fn(/* body_owner */ DefId, /* body id */ hir::BodyId),
 {
+    // NB: we use a visitor here rather than walking the keys of the
+    // hashmap so as to ensure we visit the bodies "in order".
```

Comment:
I agree, and I'd be fine sorting by `Span` or something (so the HIR visit order doesn't matter).